ui/src/websocket/websocket.ts

```diff
@@ -99,12 +113,22 @@ function useDamlStreamQuery(templateIds: string[], token?: string) {
 
     const closeWebsocket = useCallback((token: string, templateIds: string[]) => {
         return (event: CloseEvent) => {
-            // if this connection was closed unintentionally, reopen it
+            // If this connection was closed unintentionally, set it to `null` to mark for the
+            // initialization effect hook to restart it.
             if (event.code !== KEEP_CLOSED) {
-                openWebsocket(token, templateIds);
+                setActive(false);
+                if (retries <= RETRIES_ALLOWED) {
```

Comment:
The retries here is so it has a 5 second buffer between every few connection retries, otherwise it will spam connection requests.